PlasmaMVP/feeStructure.md

```diff
@@ -1,17 +1,38 @@
 # Plamsa MVP Fee Structures
 
 ## Problem:
-A Plasma validator includes a transaction in a block and corresponding confirm sigs must be added in future transactions spending the original utxo or the utxo must be withdrawn before he can actually collect transaction fees.
+There needs to be an incentive for validators to run a full node on the plasma side chain. The obvious economic incentive is to charge fees on any spent utxo. Due to the high volume of transactions per block, 2^16 slots, the fees per transactions will remain relatively low.
+
+**GOAL**: Due to the role of confirm signatures, we need to ensure that the proposer responsible for a block can securely withdraw their rightfully earned fees in the scenario of a mass exit.  
 
 ## Potential Solutions:
 
 ### Fee Budget:
-There is a mapping from address to max amount of fee you're willing to pay. Validator checks to see you haven't exceeded your fee budget before adding your transaction to the block. A validator can withdraw by showing user signed over a fee and that transaction was included in a block that the validator submitted. The validator has to submit an expensive transaction each time he wants to exit a single fee of a single transaction. This may be solved with aggregate signatures.
+We can require all participants of the side chain to allocate and bond a fee budget of Ether on the rootchain. In addition to the previous constraints on what constitutes a valid transaction, we impose that any users attempting to spend a transcation must specify a fee denomination in the txBytes that is checked against the rootchain. Per every transaction processed, the validator for a current round will deduct from all corresponding participant's fee budget. However, this opens up another order of complexity that we would like to avoid.  
+
+Does this change introduce a new challenge mechansim? Are there checks to ensure that the validator only deducts the fee amount as specified in the txBytes? How much code will this change/introduce in the rootchain?  
```

Comment:
I think the bigger issue is that this results in too many rootchain calls. A transaction on the rootchain for validator to withdraw one user's fees will result in way too many transactions.